lib/matplotlib/category.py

```diff
@@ -29,18 +29,23 @@ def convert(value, unit, axis):
         Parameters
         ----------
         value : string or iterable
-            value or list of values to be converted
-        unit : :class:`.UnitData`
-           object string unit information for value
-        axis : :class:`~matplotlib.Axis.axis`
-            axis on which the converted value is plotted
+            Value or list of values to be converted.
+        unit : `.UnitData`
+            An object mapping strings to integers.
+        axis : `~matplotlib.axis.Axis`
+            axis on which the converted value is plotted.
+
+            .. note:: *axis* is unused.
 
         Returns
         -------
-        mapped_ value : float or ndarray[float]
-
-        .. note:: axis is not used in this function
+        mapped_value : float or ndarray[float]
         """
+        if unit is None:
+            raise ValueError(
+                'Missing unit for StrCategoryConverter. This might be caused'
+                'by unintendedly mixing categorical and numeric data.')
```

Comment:
Minor preference for error messages that are single sentence and without a final dot, as that's what we mostly have in the codebase (so replace the middle dot by a semicolon).